javascript/javascript1/week4/voiceAssistant.js

```diff
@@ -0,0 +1,164 @@
+
+        const itemTodoArr = commandArr.slice(1, -3);
+
+        const todoContent = itemTodoArr.join(' ');
+
+        todo.push(todoContent);
+
+        return `${todoContent} added to your todo`;
+    }
+    else if (command === 'what is on my todo') {
+        return todo;
+    }
+    /*------------------Removing from Todo----------------------------*/
+
+    else if (command.toLowerCase().includes('remove')) {
+        const commandArr = command.toLowerCase().split(' ');
+
+        const removeTodoArr = commandArr.slice(1, -3);
+
+        todo.shift(removeTodoArr);
+
+        return `removed ${removeTodoArr} from your todo`;
+    }
+
+    /*------------------date----------------------------*/
+    else if ((command.toLowerCase().includes('today'))) {
+        const today = new Date();
+
+        const todayDate = today.getDate();
+
+        const year = today.getFullYear();
+        const currentDay = new Date().toLocaleString('en-us', { month: 'long' });
+        return `Today is ${todayDate}. of ${currentDay} ${year}`;
+    }
+    /*------------------favourite sports----------------------------*/
+
+
+    else if (command.toLowerCase().includes('my favourite sports is')) {
+        const commandArr = command.toLowerCase().split(' ');
+
+        const favSportsName = commandArr[commandArr.length - 1];
+        favsportsArr.push(favSportsName);
+        return `my favourite sports is ${favsportsArr}`;
+    }
+    else if (command.toLowerCase().includes('what is my favourite sports')) {
+        return `your favourite sport is ${favsportsArr[favsportsArr.length - 1]}.`;
+    }
+    /*------------------timer----------------------------*/
+
+    else if (command.toLowerCase().includes('set a timer for')) {
+        const commandArr = command.toLowerCase().split(' ');
+
+        const timer = commandArr[commandArr.length - 2];
+        const milliSeconds = timer * 60000; // converting 4 minutes into milliseconds 
+        setTimeout(() => {
+            console.log('Timer done')
+        }, milliSeconds);
+        return `Timer set for ${timer} minutes.`;
+    }
+
+    /*------------------Arithmetic Operations----------------------------*/
+
+    else if (command.includes('what is')) {
+        const commandArr = command.split(' ');
+        const num1 = Number(commandArr[commandArr.length - 3]);
+        const num2 = Number(commandArr[commandArr.length - 1]);
+        const operator = commandArr[commandArr.length - 2];
+        let output;
+        switch (operator) {
+            case "+":
+                return `${num1} ${operator} ${num2} = ${num1 + num2}`;
+                break;
+            case "*":
+                return `${num1} ${operator} ${num2} = ${num1 * num2}`;
+                break;
+            case "-":
+                return `${num1} ${operator} ${num2} = ${num1 - num2}`;
+                break;
+            case "/":
+                return `${num1} ${operator} ${num2} = ${(num1 / num2).toFixed(2)}`;
```

Comment:
What is this result if `num2 = 0`?

Reply:
yeah I have forgotten that case will do it shortly